bundle/edu.gemini.catalog/src/main/scala/edu/gemini/catalog/votable/VoTableParser.scala

```diff
@@ -52,8 +52,12 @@ object VoTableParser extends VoTableParser {
 sealed trait MagnitudesFilter {
   // Indicates if a field should be ignored
   def ignoredMagnitudeField(v: FieldId): Boolean = false
+  // Indicates if a magnitude is valid
   def validMagnitude(m: Magnitude): Boolean = !(m.value.isNaN || m.error.exists(_.isNaN))
-  def findBand(id: FieldId, band: String): Option[MagnitudeBand] = None
+  // Find what band the field descriptor should represent
+  def findBand(id: FieldId, band: String): Option[MagnitudeBand] = MagnitudeBand.all.find(_.name == band)
+  // filter magnitudes as a whole
+  def filterMagnitudeFields(magnitudeFields: List[(FieldId, Magnitude)]): List[Magnitude] = magnitudeFields.map(_._2).filter(validMagnitude)
```

Comment:
Admittedly not as compact but I suppose you could make one pass with `collect`, if you wanted to.

````scala
magnitudeFields.collect { case (_, mag) if validMagnitude(mag) => mag }
````